pkg/scheduler/metrics/metrics.go

```diff
@@ -173,6 +173,14 @@ var (
 			StabilityLevel: metrics.ALPHA,
 		}, []string{"type"})
 
+	unschedulableReasons = metrics.NewGaugeVec(
+		&metrics.GaugeOpts{
+			Subsystem:      SchedulerSubsystem,
+			Name:           "unschedulable_pods_by_plugin",
```

Comment:
I would recommend changing the name of the metric to `unschedulable_pods`. The fact that the metric is broken down by plugin is already explained in the Help and can be deduced by looking at the labels of the metrics so it feels redundant.